projects/pastanaga-angular/src/lib/controls/textfield/input/input-component-tests/input-component-form-control-name.spec.ts

```diff
@@ -0,0 +1,248 @@
+
+    it('should assign describedBy', fakeAsync(() => testDescribedBy(fixture, nextId)));
+
+    it('should assign readonly', fakeAsync(() => testReadonly(fixture)));
+
+    it('should focus field', fakeAsync(() => testFocus(fixture)));
+
+    it('should apply disabled state', fakeAsync(() => {
+        clearFakeAsyncZone(fixture);
+
+        fixture.componentInstance.form.disable();
+        fixture.detectChanges();
+        tick();
+        thenFieldControlIsDisabled(fixture, true);
+
+        fixture.componentInstance.form.enable();
+        fixture.detectChanges();
+        tick();
+        thenFieldControlIsDisabled(fixture, false);
+
+        if (!fixture.componentInstance.formControl) {
+            expect(true).toEqual(false);
+            return;
+        }
+
+        fixture.componentInstance.formControl.disable();
+        fixture.detectChanges();
+        tick();
+        thenFieldControlIsDisabled(fixture, true);
+
+        fixture.componentInstance.formControl.enable();
+        fixture.detectChanges();
+        tick();
+        thenFieldControlIsDisabled(fixture, false);
+    }));
+
+    it('should apply showAllErrors', fakeAsync(() => testShowAllErrors(fixture)));
+
+    it('should display errorMessages', fakeAsync(() => testErrorMessages(fixture)));
+
+    it('should not show errors when pristine', fakeAsync(() => testNoErrorForPristine(fixture)));
+
+    it('should apply errorMessage', fakeAsync(() => testErrorMessage(fixture)));
+
+    it('should assign value without emitting', fakeAsync(() => {
+        clearFakeAsyncZone(fixture);
+        jest.spyOn(fixture.componentInstance, 'onValueChange');
+        fixture.componentInstance.form.setValue({ text: 'test' });
+        fixture.detectChanges();
+        tick();
+        thenFieldControlHasValue(fixture, 'test');
+        expect(fixture.componentInstance.onValueChange).toHaveReturnedTimes(0);
+
+        if (!fixture.componentInstance.formControl) {
+            expect(true).toEqual(false);
+            return;
+        }
+        fixture.componentInstance.formControl.setValue('test2');
+        fixture.detectChanges();
+        tick();
+        thenFieldControlHasValue(fixture, 'test2');
+        expect(fixture.componentInstance.onValueChange).toHaveReturnedTimes(0);
+    }));
+
+    it('should assign type', fakeAsync(() => testType(fixture)));
+
+    it('should assign placeholder', fakeAsync(() => testPlaceholder(fixture)));
+
+    it('should validate required', fakeAsync(() => testRequired(fixture)));
+
+    it('should validate pattern', fakeAsync(() => testPattern(fixture)));
+
+    it('should validate min', fakeAsync(() => testMin(fixture)));
+
+    it('should validate max', fakeAsync(() => testMax(fixture)));
+
+    it('should validate maxlength', fakeAsync(() => testMaxlength(fixture)));
+
+    it('should assign noAutoComplete', fakeAsync(() => testAutocomplete(fixture)));
+
+    it('should apply acceptHtmlTags', fakeAsync(() => {
```

Comment:
why didn't you use the helper function you created for acceptHtmlTags?

Reply:
here the value is updated using the formControl... I didn't write helper function for formControls specifics but I will